test/IntegrationTests/MSTest.Acceptance.IntegrationTests/AssemblyAndClassInitializeTimeout.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Testing.Platform.Acceptance.IntegrationTests;
+using Microsoft.Testing.Platform.Acceptance.IntegrationTests.Helpers;
+
+namespace MSTest.Acceptance.IntegrationTests;
+
+[TestGroup]
+public class AssemblyAndClassInitializeTimeout : AcceptanceTestBase
+{
+    private readonly string _assetName = "AssemblyAndClassInitializeTimeout";
+    private readonly AcceptanceFixture _acceptanceFixture;
+
+    public AssemblyAndClassInitializeTimeout(ITestExecutionContext testExecutionContext, AcceptanceFixture acceptanceFixture)
+        : base(testExecutionContext)
+    {
+        _acceptanceFixture = acceptanceFixture;
+    }
+
+    [ArgumentsProvider(nameof(GetBuildMatrixTfmBuildConfiguration))]
+    public async Task AssemblyInitialize_Timeout_Should_Stop_Execution(string tfm, BuildConfiguration buildConfiguration)
+    {
+        using TestAsset testAsset = await TestAsset.GenerateAssetAsync(_assetName, SourceCode
+            .PatchCodeWithReplace("$TargetFramework$", tfm)
+            .PatchCodeWithReplace("$MSTestVersion$", MSTestVersion), addPublicFeeds: true);
```

Comment:
I don't think this is needed

```suggestion
            .PatchCodeWithReplace("$MSTestVersion$", MSTestVersion));
```